client/src/components/CollegeShortlist.js

```diff
@@ -0,0 +1,100 @@
+import React, { Component } from 'react';
+import { Menu, Label, Card, Button, Icon } from 'semantic-ui-react';
+import CollegeShortlistModal from './CollegeShortlistModal';
+
+const STUDENT = "STUDENT"
+
+export default class CollegeShortlist extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            activeItem: "shortlist",
+            collegeModalOpen: false,
+            collegeShortlist: [],
+        }
+        this.openCollegeModal = this.openCollegeModal.bind(this);
+        this.closeCollegeModal = this.closeCollegeModal.bind(this);
+        this.addToShortlist = this.addToShortlist.bind(this);
+    }
+
+    openCollegeModal() {
+        this.setState({
+            collegeModalOpen: true
+        })
+    }
+    closeCollegeModal() {
+        this.setState({
+            collegeModalOpen: false
+        }, () => {
+            this.props.refreshProfile(STUDENT, this.props.details._id)
+        })
+    }
+
+    addToShortlist(newCollege) {        
+        let currentCollegeShortlist = this.props.details.collegeShortlist
+        currentCollegeShortlist.push(newCollege) 
+        this.setState({
+            shortlist: currentCollegeShortlist
+        })
+        this.state.collegeShortlist.push(newCollege);
```

Comment:
... and won't need this ([directly changing `this.state`](https://daveceddia.com/why-not-modify-react-state-directly/#:~:text=Mutating%20state%20directly%20can%20lead,its%20parent%20re%2Drenders).) isn't advisable)